src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/PortManager.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace WcfTestBridgeCommon
+{
+    // This class exists to create and delete firewall rules to
+    // manage which ports are open on behalf of the Bridge.
+    public class PortManager
+    {
+        private static HashSet<int> s_inPorts = new HashSet<int>();
+        private static HashSet<int> s_outPorts = new HashSet<int>();
+        private static object s_portLock = new object();
+
+        public static void OpenPortInFirewall(int port)
+        {
+            OpenPortInFirewall(port, "in", s_inPorts);
+            OpenPortInFirewall(port, "out", s_outPorts);
```

Comment:
You shouldn't need to add the "out" rules as the bridge isn't making outgoing requests.